cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3030,25 +3051,48 @@ TEST(ArrowReadWrite, FixedSizeList) {
   CheckSimpleRoundtrip(table, 2, props_store_schema);
 }
 
-TEST(ArrowReadWrite, ListOfStructOfList2) {
+TEST(ArrowReadWrite, FixedSizeList2) {
   using ::arrow::field;
-  using ::arrow::list;
+  using ::arrow::fixed_size_list;
   using ::arrow::struct_;
 
-  auto type =
-      list(field("item",
-                 struct_({field("a", ::arrow::int16(), /*nullable=*/false),
-                          field("b", list(::arrow::int64()), /*nullable=*/false)}),
-                 /*nullable=*/false));
+  auto type = fixed_size_list(::arrow::int16(), /*size=*/3);
 
   const char* json = R"([
-      [{"a": 123, "b": [1, 2, 3]}],
       null,
-      [],
-      [{"a": 456, "b": []}, {"a": 789, "b": [null]}, {"a": 876, "b": [4, 5, 6]}]])";
+      [null, 1, 2],
+      [null, 3, 4],
+      null,
+      [5, null, 6],
+      [7, null, 8],
+      null,
+      [9, 10, null],
+      null,
+      null,
+      [11, 12, 13],
+      [14, 15, 16]])";
   auto array = ::arrow::ArrayFromJSON(type, json);
   auto table = ::arrow::Table::Make(::arrow::schema({field("root", type)}), {array});
-  CheckSimpleRoundtrip(table, 2);
+  auto props_store_schema = ArrowWriterProperties::Builder().store_schema()->build();
+  CheckSimpleRoundtrip(table, 2, props_store_schema);
```

Comment:
I think we need more tests here that can cover nulls in the middle and multiple levels of nested fixed sizes lists with other types of lists.